component/distributed_cache/distributed_cache.go

```diff
@@ -109,20 +113,126 @@ func (dc *DistributedCache) Stop() error {
-func (distributedCache *DistributedCache) OnConfigChange() {
+func (dc *DistributedCache) OnConfigChange() {
+}
+
+func (dc *DistributedCache) GetAttr(options internal.GetAttrOptions) (*internal.ObjAttr, error) {
+	ok, attr := isPlaceholderDirForRoot(options.Name)
+	if ok {
+		return attr, nil
+	}
+	if !isPathValid(options.Name) {
+		return nil, syscall.ENOENT
+	}
+	if options.Name == "__CACHE__"+dc.cacheID {
+		return nil, syscall.ENOENT
+	}
+	newPath := options.Name
+	// Check if properties should be fetched from Azure
+	getAttrFromAzure, resPath := isPathContainsAzureVirtualComponent(options.Name)
+	if getAttrFromAzure {
+		log.Info("DistributedCache::GetAttr : Path is having Azure subcomponent, path : %s", options.Name)
+		newPath = resPath
+	}
+	// Check if properties should be fetched from Dcache
+	getAttrFromDcache, resPath := isPathContainsDcacheVirtualComponent(options.Name)
+	if getAttrFromDcache {
+		log.Info("DistributedCache::GetAttr : Path is having Dcache subcomponent, path : %s", options.Name)
+		newPath = filepath.Join("__CACHE__"+dc.cacheID+"/Objects", resPath)
+		// Check for existense of the directory here as it's path dont end with .md
+		attr, err := dc.NextComponent().GetAttr(internal.GetAttrOptions{Name: newPath})
+		if err == nil {
+			return attr, nil
+		}
+		// Now the directory is not present, check for the existence of the file
+		newPath += ".md"
+	}
+	// Default is to get the properties from the Azure
+	attr, err := dc.NextComponent().GetAttr(internal.GetAttrOptions{Name: newPath})
+	if err != nil {
+		return nil, err
+	}
+	// Modify the attr if it came from specific virtual component.
+	if getAttrFromAzure || getAttrFromDcache {
+		attr.Path = options.Name
+		attr.Name = filepath.Base(options.Name)
+	}
+	return attr, nil
+}
+
+func (dc *DistributedCache) StreamDir(options internal.StreamDirOptions) ([]*internal.ObjAttr, string, error) {
+	// Check If directory has azure as virtual subcomponent
+	getListFromAzure, resPath := isPathContainsAzureVirtualComponent(options.Name)
+	if getListFromAzure {
+		log.Info("DistributedCache::StreamDir : Path is having Azure subcomponent, path : %s", options.Name)
+		options.Name = resPath
+	}
+	// Check If directory has dcache as virtual subcomponent
+	getListFromDcache, resPath := isPathContainsDcacheVirtualComponent(options.Name)
+	if getListFromDcache {
+		log.Info("DistributedCache::StreamDir : Path is having Dcache subcomponent, path : %s", options.Name)
+		options.Name = filepath.Join("__CACHE__"+dc.cacheID+"/Objects", resPath)
+	}
+	// Normal listing.
+	dirList, token, err := dc.NextComponent().StreamDir(options)
+	if err != nil {
+		return dirList, token, err
+	}
+	if getListFromDcache {
+		// Remove the .md suffixes from the filenames.
+		for _, attr := range dirList {
+			attr.Name = strings.TrimSuffix(attr.Name, ".md")
+		}
+	}
+	if isMountPointRoot(options.Name) {
+		dirList = hideCacheFolder(dirList, "__CACHE__"+dc.cacheID)
+	}
```

Comment:
we should not have ``__CACH__<cacheid>`` in the enumerated result, why do we need to hide this

Reply:
nvm, since we could be enumerating the container root we may get the ``__CACHE__`` dir.
As mentioned above, let's hide anything starting with ``__CACHE__``.
we can call this hideCacheMetadata()